### ISSUE TITLE
Publisher not closing tcp connections

### ISSUE THREAD

**Issue by Sniffq** [intelsdi-x/snap-plugin-publisher-influxdb#60]
I'm trying to test publising some stats to InfluxDB v1.0 on Centos 7.
The setup is from the [Getting Started Tutorial](https://github.com/intelsdi-x/snap#getting-started)
I start snapd 
```sh
sudo nohup /usr/local/bin/snapd --config myconfig.json &
```
```json
{
    "log_level": 1,
    "log_path": "/var/log/snap",
    "gomaxprocs": 2,
    "control": {
        "auto_discover_path": "/home/centos/snap/plugins_use",
        "plugin_trust_level": 0,
        "plugins": {
            "all": {
                "user": "none",
                "log-level": "debug",
                "password": "none"
            },
            "collector": {
                "all": {
                },
                "psutil": {
                    "all": {
                        "path": "/usr/local/bin/psutil"
                    }
                }
            },
            "publisher": {
            }
        }
    }
}
```
with snap-plugin-collector-psutil and snap-plugin-publisher-influxdb loaded, and then I create a task with a following config:
```sh
snapctl task create -t  tasks/psutil-influx.json
```
```json
{
    "version": 1,
    "schedule": {
        "type": "simple",
        "interval": "1s"
    },
    "workflow": {
        "collect": {
            "metrics": {
                "/intel/psutil/load/load1": {},
                "/intel/psutil/load/load5": {},
                "/intel/psutil/load/load15": {},
                "/intel/psutil/vm/available": {},
                "/intel/psutil/vm/free": {},
                "/intel/psutil/vm/used": {}
            },
            "process": null,
            "publish": [
                {
                    "plugin_name": "influx",
                    "config": {
                        "log-level" : "debug",
                        "version": 13,
                        "host": "127.0.0.1",
                        "port": 8086,
                        "database": "db_test",
                        "user": "admin",
                        "password": "password"
                    }
                }
            ]
        }
    }
}
```
The task starts successfully an begins to push data to InfluxDB, but the number of TCP connections continue to grow fast, until I stop the task, after that they return to initial number more or less.
Initially I got an error "too many open files" like in this issue #51, so I increased the ulimit, but the problem is connection are not getting closed and I can see tens of thousands of them after 30 minutes.  

```
[centos@vm snap]$ sudo lsof |grep influx | wc -l
383
[centos@vm snap]$ snapctl task create -t  tasks/psutil-influx.json
Using task manifest to create task
Task created
ID: d7d26338-44e4-4888-adaa-be7e50d79f45
Name: Task-d7d26338-44e4-4888-adaa-be7e50d79f45
State: Running
[centos@vm snap]$ snapctl task list
ID 					 NAME 						 STATE 		 HIT 	 MISS 	 FAIL 	 CREATED 		 LAST FAILURE
d7d26338-44e4-4888-adaa-be7e50d79f45 	 Task-d7d26338-44e4-4888-adaa-be7e50d79f45 	 Running 	 3 	 0 	 0 	 9:46PM 6-29-2016 	 
[centos@vm snap]$ snapctl task list
ID 					 NAME 						 STATE 		 HIT 	 MISS 	 FAIL 	 CREATED 		 LAST FAILURE
d7d26338-44e4-4888-adaa-be7e50d79f45 	 Task-d7d26338-44e4-4888-adaa-be7e50d79f45 	 Running 	 6 	 0 	 0 	 9:46PM 6-29-2016 	 
[centos@vm snap]$ sudo lsof |grep influx | wc -l
535
[centos@vm snap]$ sudo lsof |grep influx | wc -l
1765
[centos@vm snap]$ sudo lsof |grep influx | wc -l
17655
[...]
[centos@vm snap]$ snapctl task stop d7d26338-44e4-4888-adaa-be7e50d79f45
Task stopped:
ID: d7d26338-44e4-4888-adaa-be7e50d79f45
[centos@vm snap]$  sudo lsof |grep influx | wc -l
395
```

```
[centos@vm snap]$ sudo lsof |grep influx
[...]
influxd   21061 21066 influxdb 1011u     IPv6             221017       0t0        TCP localhost:d-s-n->localhost:37671 (ESTABLISHED)
influxd   21061 21066 influxdb 1012u     IPv6             221019       0t0        TCP localhost:d-s-n->localhost:37672 (ESTABLISHED)
influxd   21061 21066 influxdb 1013u     IPv6             207517       0t0        TCP localhost:d-s-n->localhost:37673 (ESTABLISHED)
influxd   21061 21066 influxdb 1014u     IPv6             221021       0t0        TCP localhost:d-s-n->localhost:37674 (ESTABLISHED)
influxd   21061 21066 influxdb 1015u     IPv6             207520       0t0        TCP localhost:d-s-n->localhost:37675 (ESTABLISHED)
influxd   21061 21066 influxdb 1016u     IPv6             207533       0t0        TCP localhost:d-s-n->localhost:37679 (ESTABLISHED)
```

Also I can see some logging in the connection pool code [(this commit):](https://github.com/lynxbat/snap-plugin-publisher-influxdb/commit/f4a6fda139414d43f96b1ade45df67adadc424e6)
```go
logger.Debug("Opening new InfluxDB connection[", user, "@", db, " ", u.String(), "]")
[...]
logger.Debug("Using open InfluxDB connection[", user, "@", db, " ", u.String(), "]")
```

 but I cannot find where the log is being written, nothing about opening or closing connections in  /var/log/snap/snapd.log

**Comment by lynxbat** [comment]
@Sniffq thanks for this. We recently added some recycling logic and must have introduced a bug.

**Comment by Sniffq** [comment]
@lynxbat Maybe you could post a sample config for the publisher or give me a hint where to look for the log or debug messages so I could investigate it?

**Comment by lynxbat** [comment]
Logs for plugins are separate (right now) from snapd logs. Looks inside /tmp and see if it is writing there.

**Comment by jose5918** [comment]
I am also having this issue on Ubuntu 14.04.3.  I have confirmed the bug using `sudo lsof |grep influx | wc -l` . I am working on getting some debug logging from it right now and I will post it when I have it.

**Comment by jose5918** [comment]
I looked around in the plugin logs for influx while running snap in debug and couldn't find any information about open files. The only error was in the snap log 
```
Task disabled with error: Publish call error: Post http://ec2-54-197-0-97.compute-1.amazonaws.com:8086/write?consistency=&db=snap&precision=s&rp=default: dial tcp: lookup ec2-54-197-0-97.compute-1.amazonaws.com on 172.31.0.2:53: dial udp 172.31.0.2:53: socket: too many open files
```

**Comment by lynxbat** [comment]
The log level defaults to WARN and most of the DEBUG is not going to print.

**Comment by Sniffq** [comment]
@jose5918 Yep, had the same message and task disabled before I raised the ulimit from 1024 to 65536. 
@lynxbat Do I set the log level right way in my task .json? 
```json
"config": {
   "log-level": "debug"
...
```

**Comment by lynxbat** [comment]
I think it may need to be `DEBUG` or `1`. Trying to find the doc on it. It is a portion we are working a unifying shortly.

**Comment by lynxbat** [comment]
Actually, it looks like the wiring to use this key was removed in a version past.

Best to set the default to debug in the plugin and compile: https://github.com/intelsdi-x/snap-plugin-publisher-influxdb/blob/master/influx/influx.go#L252

**Comment by Sniffq** [comment]
Will try, thanks!

**Comment by Sniffq** [comment]
@lynxbat Found logs in /tmp:
```sh
snap-plugin-publisher-influxdb.stderr
snap-plugin-publisher-influxdb.stdout
snap-plugin-publisher-influxdb.log
```
The debug output is being dumped into /tmp/snap-plugin-publisher-influxdb.stdout, and I can see the publisher opening new connection every time:

```log
2016/06/30 12:30:38 time="2016-06-30T12:30:38+03:00" level=debug msg="Opening new InfluxDB connection[admin@db_test http://127.0.0.1:8086]" plugin-name=influx plugin-type=publisher plugin-version=13 
2016/06/30 12:30:38 time="2016-06-30T12:30:38+03:00" level=debug msg="publishing metrics" batch-points=[intel/psutil/vm/free,source=server.localdomain,unit=B value=902127616i 1467279038263723729 intel/psutil/vm/used,source=server.localdomain,unit=B value=794062848i 1467279038263837851 intel/psutil/load/load1,source=server.localdomain,unit=Load/1M value=0 1467279038263886871 intel/psutil/load/load15,source=server.localdomain,unit=Load/15M value=0.05 1467279038263916312 intel/psutil/load/load5,source=server.localdomain,unit=Load/5M value=0.01 1467279038263948836 intel/psutil/vm/available,source=server.localdomain,unit=B value=3181182976i 1467279038264057325] plugin-name=influx plugin-type=publisher plugin-version=13 
2016/06/30 12:30:40 time="2016-06-30T12:30:40+03:00" level=debug msg="Opening new InfluxDB connection[admin@db_test http://127.0.0.1:8086]" plugin-name=influx plugin-type=publisher plugin-version=13 
2016/06/30 12:30:40 time="2016-06-30T12:30:40+03:00" level=debug msg="publishing metrics" batch-points=[intel/psutil/vm/free,source=server.localdomain,unit=B value=899338240i 1467279040263410947 intel/psutil/vm/used,source=server.localdomain,unit=B value=796610560i 1467279040263538121 intel/psutil/load/load1,source=server.localdomain,unit=Load/1M value=0 1467279040263574490 intel/psutil/load/load15,source=server.localdomain,unit=Load/15M value=0.05 1467279040263609343 intel/psutil/load/load5,source=server.localdomain,unit=Load/5M value=0.01 1467279040263635384 intel/psutil/vm/available,source=server.localdomain,unit=B value=3178635264i 1467279040263753437] plugin-name=influx plugin-type=publisher plugin-version=13 
2016/06/30 12:30:42 time="2016-06-30T12:30:42+03:00" level=debug msg="Opening new InfluxDB connection[admin@db_test http://127.0.0.1:8086]" plugin-name=influx plugin-type=publisher plugin-version=13 
2016/06/30 12:30:42 time="2016-06-30T12:30:42+03:00" level=debug msg="publishing metrics" batch-points=[intel/psutil/vm/free,source=server.localdomain,unit=B value=899633152i 1467279042263563159 intel/psutil/vm/used,source=server.localdomain,unit=B value=796307456i 1467279042263673597 intel/psutil/load/load1,source=server.localdomain,unit=Load/1M value=0 1467279042263717532 intel/psutil/load/load15,source=server.localdomain,unit=Load/15M value=0.05 1467279042263744698 intel/psutil/load/load5,source=server.localdomain,unit=Load/5M value=0.01 1467279042263769770 intel/psutil/vm/available,source=server.localdomain,unit=B value=3178938368i 1467279042263860669] plugin-name=influx plugin-type=publisher plugin-version=13 
2016/06/30 12:30:44 time="2016-06-30T12:30:44+03:00" level=debug msg="Opening new InfluxDB connection[admin@db_test http://127.0.0.1:8086]" plugin-name=influx plugin-type=publisher plugin-version=13 
2016/06/30 12:30:44 time="2016-06-30T12:30:44+03:00" level=debug msg="publishing metrics" batch-points=[intel/psutil/vm/free,source=server.localdomain,unit=B value=898695168i 1467279044265641030 intel/psutil/vm/used,source=server.localdomain,unit=B value=797192192i 1467279044265767428 intel/psutil/load/load1,source=server.localdomain,unit=Load/1M value=0 1467279044265806248 intel/psutil/load/load15,source=server.localdomain,unit=Load/15M value=0.05 1467279044265832565 intel/psutil/load/load5,source=server.localdomain,unit=Load/5M value=0.01 1467279044265859688 intel/psutil/vm/available,source=server.localdomain,unit=B value=3178053632i 1467279044266004271] plugin-name=influx plugin-type=publisher plugin-version=13 
2016/06/30 12:30:46 time="2016-06-30T12:30:46+03:00" level=debug msg="Opening new InfluxDB connection[admin@db_test http://127.0.0.1:8086]" plugin-name=influx plugin-type=publisher plugin-version=13 
```

**Comment by Sniffq** [comment]
@lynxbat I've added some debug and it seems like the problem was in connectionPool check -  if connPool[u.String()] == nil was using u.String() as a key while everywhere else there is a composite key used, so this check always evaluated to false and new connections were created instead of reusing existing ones.
```go
key := connectionKey(u, user, db)
...
        // Do we have a existing client?
        if connPool[u.String()] == nil {
...
connPool[key] = cCon
```

```log
2016/06/30 12:59:07 time="2016-06-30T12:59:07+03:00" level=debug msg="Selecting Client connection, the connectionKey is: http://127.0.0.1:8086:admin:db_test u=http://127.0.0.1:8086 connPool[u.String()]=<nil> connPool[key]=<nil>" plugin-name=influx plugin-type=publisher plugin-version=13 
2016/06/30 12:59:07 time="2016-06-30T12:59:07+03:00" level=debug msg="Opening new InfluxDB connection[admin@db_test http://127.0.0.1:8086]" plugin-name=influx plugin-type=publisher plugin-version=13 
2016/06/30 12:59:07 time="2016-06-30T12:59:07+03:00" level=debug msg="publishing metrics" batch-points=[intel/psutil/vm/used,source=server.localdomain,unit=B value=802725888i 1467280747364305121 intel/psutil/load/load1,source=server.localdomain,unit=Load/1M value=0.14 1467280747364357195 intel/psutil/load/load15,source=server.localdomain,unit=Load/15M value=0.05 1467280747364385728 intel/psutil/load/load5,source=server.localdomain,unit=Load/5M value=0.08 1467280747364420581 intel/psutil/vm/available,source=server.localdomain,unit=B value=3172519936i 1467280747364525668 intel/psutil/vm/free,source=server.localdomain,unit=B value=890425344i 1467280747364631430] plugin-name=influx plugin-type=publisher plugin-version=13 
2016/06/30 12:59:09 time="2016-06-30T12:59:09+03:00" level=debug msg="Selecting Client connection, the connectionKey is: http://127.0.0.1:8086:admin:db_test u=http://127.0.0.1:8086 connPool[u.String()]=<nil> connPool[key]=&{http://127.0.0.1:8086:admin:db_test 0xc82013bd10 2016-06-30 12:59:07.367327718 +0300 MSK}" plugin-name=influx plugin-type=publisher plugin-version=13 
2016/06/30 12:59:09 time="2016-06-30T12:59:09+03:00" level=debug msg="Opening new InfluxDB connection[admin@db_test http://127.0.0.1:8086]" plugin-name=influx plugin-type=publisher plugin-version=13 
2016/06/30 12:59:09 time="2016-06-30T12:59:09+03:00" level=debug msg="publishing metrics" batch-points=[intel/psutil/vm/used,source=server.localdomain,unit=B value=803590144i 1467280749363557176 intel/psutil/load/load1,source=server.localdomain,unit=Load/1M value=0.14 1467280749363594908 intel/psutil/load/load15,source=server.localdomain,unit=Load/15M value=0.05 1467280749363620927 intel/psutil/load/load5,source=server.localdomain,unit=Load/5M value=0.08 1467280749363654532 intel/psutil/vm/available,source=server.localdomain,unit=B value=3171655680i 1467280749363751742 intel/psutil/vm/free,source=server.localdomain,unit=B value=889544704i 1467280749363880627] plugin-name=influx plugin-type=publisher plugin-version=13 
2016/06/30 12:59:11 time="2016-06-30T12:59:11+03:00" level=debug msg="Selecting Client connection, the connectionKey is: http://127.0.0.1:8086:admin:db_test u=http://127.0.0.1:8086 connPool[u.String()]=<nil> connPool[key]=&{http://127.0.0.1:8086:admin:db_test 0xc8203f51f0 2016-06-30 12:59:09.366298022 +0300 MSK}" plugin-name=influx plugin-type=publisher plugin-version=13 
2016/06/30 12:59:11 time="2016-06-30T12:59:11+03:00" level=debug msg="Opening new InfluxDB connection[admin@db_test http://127.0.0.1:8086]" plugin-name=influx plugin-type=publisher plugin-version=13 
2016/06/30 12:59:11 time="2016-06-30T12:59:11+03:00" level=debug msg="publishing metrics" batch-points=[intel/psutil/vm/used,source=server.localdomain,unit=B value=803672064i 1467280751364784395 intel/psutil/load/load1,source=server.localdomain,unit=Load/1M value=0.21 1467280751364838683 intel/psutil/load/load15,source=server.localdomain,unit=Load/15M value=0.06 1467280751364870787 intel/psutil/load/load5,source=server.localdomain,unit=Load/5M value=0.09 1467280751364895982 intel/psutil/vm/available,source=server.localdomain,unit=B value=3171573760i 1467280751365054570 intel/psutil/vm/free,source=server.localdomain,unit=B value=889454592i 1467280751365170713] plugin-name=influx plugin-type=publisher plugin-version=13 
2016/06/30 12:59:13 time="2016-06-30T12:59:13+03:00" level=debug msg="Selecting Client connection, the connectionKey is: http://127.0.0.1:8086:admin:db_test u=http://127.0.0.1:8086 connPool[u.String()]=<nil> connPool[key]=&{http://127.0.0.1:8086:admin:db_test 0xc8203fd030 2016-06-30 12:59:11.367352333 +0300 MSK}" plugin-name=influx plugin-type=publisher plugin-version=13 
2016/06/30 12:59:13 time="2016-06-30T12:59:13+03:00" level=debug msg="Opening new InfluxDB connection[admin@db_test http://127.0.0.1:8086]" plugin-name=influx plugin-type=publisher plugin-version=13 
2016/06/30 12:59:13 time="2016-06-30T12:59:13+03:00" level=debug msg="publishing metrics" batch-points=[intel/psutil/vm/used,source=server.localdomain,unit=B value=803696640i 1467280753366066363 intel/psutil/load/load1,source=server.localdomain,unit=Load/1M value=0.21 1467280753366118376 intel/psutil/load/load15,source=server.localdomain,unit=Load/15M value=0.06 1467280753366154669 intel/psutil/load/load5,source=server.localdomain,unit=Load/5M value=0.09 1467280753366187628 intel/psutil/vm/available,source=server.localdomain,unit=B value=3171549184i 1467280753366750259 intel/psutil/vm/free,source=server.localdomain,unit=B value=889421824i 1467280753366954706] plugin-name=influx plugin-type=publisher plugin-version=13 
2016/06/30 12:59:15 time="2016-06-30T12:59:15+03:00" level=debug msg="Selecting Client connection, the connectionKey is: http://127.0.0.1:8086:admin:db_test u=http://127.0.0.1:8086 connPool[u.String()]=<nil> connPool[key]=&{http://127.0.0.1:8086:admin:db_test 0xc82042e4e0 2016-06-30 12:59:13.369251374 +0300 MSK}" plugin-name=influx plugin-type=publisher plugin-version=13 
2016/06/30 12:59:15 time="2016-06-30T12:59:15+03:00" level=debug msg="Opening new InfluxDB connection[admin@db_test http://127.0.0.1:8086]" plugin-name=influx plugin-type=publisher plugin-version=13 
```

I changed u.String() to key variable and it seems to fix the issue for me, you can see that is is the publisher is "Selecting Client connection" now:

```log
2016/06/30 13:02:52 time="2016-06-30T13:02:52+03:00" level=debug msg="Selecting Client connection, the connectionKey is: http://127.0.0.1:8086:admin:db_test u=http://127.0.0.1:8086 connPool[u.String()]=<nil> connPool[key]=<nil>" plugin-name=influx plugin-type=publisher plugin-version=13
2016/06/30 13:02:52 time="2016-06-30T13:02:52+03:00" level=debug msg="Opening new InfluxDB connection[admin@db_test http://127.0.0.1:8086]" plugin-name=influx plugin-type=publisher plugin-version=13
2016/06/30 13:02:53 time="2016-06-30T13:02:53+03:00" level=debug msg="publishing metrics" batch-points=[intel/psutil/load/load1,source=server.localdomain,unit=Load/1M value=0.21 1467280972875633961 intel/psutil/load/load15,source=server.localdomain,unit=Load/15M value=0.08 1467280972875678544 intel/psutil/load/load5,source=server.localdomain,unit=Load/5M value=0.14 1467280972875706748 intel/psutil/vm/available,source=server.localdomain,unit=B value=3169492992i 1467280972875863006 intel/psutil/vm/free,source=server.localdomain,unit=B value=886976512i 1467280972875964046 intel/psutil/vm/used,source=server.localdomain,unit=B value=805752832i 1467280972876093471] plugin-name=influx plugin-type=publisher plugin-version=13
2016/06/30 13:02:54 time="2016-06-30T13:02:54+03:00" level=debug msg="Selecting Client connection, the connectionKey is: http://127.0.0.1:8086:admin:db_test u=http://127.0.0.1:8086 connPool[u.String()]=<nil> connPool[key]=&{http://127.0.0.1:8086:admin:db_test 0xc8204701e0 2016-06-30 13:02:52.878991011 +0300 MSK}" plugin-name=influx plugin-type=publisher plugin-version=13
2016/06/30 13:02:54 time="2016-06-30T13:02:54+03:00" level=debug msg="Using open InfluxDB connection[admin@db_test http://127.0.0.1:8086]" plugin-name=influx plugin-type=publisher plugin-version=13
2016/06/30 13:02:54 time="2016-06-30T13:02:54+03:00" level=debug msg="publishing metrics" batch-points=[intel/psutil/load/load1,source=server.localdomain,unit=Load/1M value=0.19 1467280974876922977 intel/psutil/load/load15,source=server.localdomain,unit=Load/15M value=0.08 1467280974876954283 intel/psutil/load/load5,source=server.localdomain,unit=Load/5M value=0.14 1467280974876980032 intel/psutil/vm/available,source=server.localdomain,unit=B value=3169206272i 1467280974877177247 intel/psutil/vm/free,source=server.localdomain,unit=B value=886689792i 1467280974877284961 intel/psutil/vm/used,source=server.localdomain,unit=B value=806039552i 1467280974877386872] plugin-name=influx plugin-type=publisher plugin-version=13
2016/06/30 13:02:56 time="2016-06-30T13:02:56+03:00" level=debug msg="Selecting Client connection, the connectionKey is: http://127.0.0.1:8086:admin:db_test u=http://127.0.0.1:8086 connPool[u.String()]=<nil> connPool[key]=&{http://127.0.0.1:8086:admin:db_test 0xc8204701e0 2016-06-30 13:02:54.879488939 +0300 MSK}" plugin-name=influx plugin-type=publisher plugin-version=13
2016/06/30 13:02:56 time="2016-06-30T13:02:56+03:00" level=debug msg="Using open InfluxDB connection[admin@db_test http://127.0.0.1:8086]" plugin-name=influx plugin-type=publisher plugin-version=13
2016/06/30 13:02:56 time="2016-06-30T13:02:56+03:00" level=debug msg="publishing metrics" batch-points=[intel/psutil/load/load1,source=server.localdomain,unit=Load/1M value=0.19 1467280976875608365 intel/psutil/load/load15,source=server.localdomain,unit=Load/15M value=0.08 1467280976875638813 intel/psutil/load/load5,source=server.localdomain,unit=Load/5M value=0.14 1467280976875665229 intel/psutil/vm/available,source=server.localdomain,unit=B value=3168636928i 1467280976875811538 intel/psutil/vm/free,source=server.localdomain,unit=B value=886116352i 1467280976875911635 intel/psutil/vm/used,source=server.localdomain,unit=B value=806608896i 1467280976876033952] plugin-name=influx plugin-type=publisher plugin-version=13
2016/06/30 13:02:58 time="2016-06-30T13:02:58+03:00" level=debug msg="Selecting Client connection, the connectionKey is: http://127.0.0.1:8086:admin:db_test u=http://127.0.0.1:8086 connPool[u.String()]=<nil> connPool[key]=&{http://127.0.0.1:8086:admin:db_test 0xc8204701e0 2016-06-30 13:02:56.878035765 +0300 MSK}" plugin-name=influx plugin-type=publisher plugin-version=13
2016/06/30 13:02:58 time="2016-06-30T13:02:58+03:00" level=debug msg="Using open InfluxDB connection[admin@db_test http://127.0.0.1:8086]" plugin-name=influx plugin-type=publisher plugin-version=13
2016/06/30 13:02:58 time="2016-06-30T13:02:58+03:00" level=debug msg="publishing metrics" batch-points=[intel/psutil/load/load1,source=server.localdomain,unit=Load/1M value=0.19 1467280978875582306 intel/psutil/load/load15,source=server.localdomain,unit=Load/15M value=0.08 1467280978875613855 intel/psutil/load/load5,source=server.localdomain,unit=Load/5M value=0.14 1467280978875651742 intel/psutil/vm/available,source=server.localdomain,unit=B value=3168612352i 1467280978875786127 intel/psutil/vm/free,source=server.localdomain,unit=B value=886083584i 1467280978875882384 intel/psutil/vm/used,source=server.localdomain,unit=B value=806633472i 1467280978875978581] plugin-name=influx plugin-type=publisher plugin-version=13
2016/06/30 13:03:00 time="2016-06-30T13:03:00+03:00" level=debug msg="Selecting Client connection, the connectionKey is: http://127.0.0.1:8086:admin:db_test u=http://127.0.0.1:8086 connPool[u.String()]=<nil> connPool[key]=&{http://127.0.0.1:8086:admin:db_test 0xc8204701e0 2016-06-30 13:02:58.877908125 +0300 MSK}" plugin-name=influx plugin-type=publisher plugin-version=13
```

The number of connections also seem not to grow when creating the task with the same config as earlier:
```sh
[centos@server snap]$ sudo nohup /usr/local/bin/snapd --config myconfig.json &
snapctl task create -t  tasks/psutil-influx.json
Using task manifest to create task
Task created
ID: 1c42bf72-3ff2-46ad-87c6-8168b2035be3
Name: Task-1c42bf72-3ff2-46ad-87c6-8168b2035be3
State: Running
[centos@server snap]$ sudo nohup /usr/local/bin/snapd --config myconfig.json &^C
[centos@server snap]$ snapctl task list
ID 					 NAME 						 STATE 		 HIT 	 MISS 	 FAIL 	 CREATED 		 LAST FAILURE
1c42bf72-3ff2-46ad-87c6-8168b2035be3 	 Task-1c42bf72-3ff2-46ad-87c6-8168b2035be3 	 Running 	 4 	 0 	 0 	 1:02PM 6-30-2016 	 
[centos@server snap]$ less /tmp/snap-plugin-publisher-influxdb.stdout
[centos@server snap]$ sudo lsof |grep influx | wc -l
523
[centos@server snap]$ sudo lsof |grep influx | wc -l
525
... after a while
[centos@server snap]$ sudo lsof |grep influx | wc -l
525
```

I've created a pull request for that #61

**Comment by lynxbat** [comment]
Fixed by #61
